benchmarks/bench_gap_divergence.py

```diff
@@ -0,0 +1,201 @@
+            # Loop over batches
+            for i, (unq_idx, idx) in enumerate(batch_lookup(lookup, n=self.batch_size)):
+                if i == n_batch - 1:
+                    W_last = self.W_.copy()
+                # Update activations unq_H
+                unq_H[unq_idx] = _multiplicative_update_h(
+                    unq_V[unq_idx],
+                    self.W_,
+                    unq_H[unq_idx],
+                    epsilon=1e-3,
+                    max_iter=self.max_iter_e_step,
+                    rescale_W=self.rescale_W,
+                    gamma_shape_prior=self.gamma_shape_prior,
+                    gamma_scale_prior=self.gamma_scale_prior,
+                )
+                # Update the topics self.W_
+                _multiplicative_update_w(
+                    unq_V[idx],
+                    self.W_,
+                    self.A_,
+                    self.B_,
+                    unq_H[idx],
+                    self.rescale_W,
+                    self.rho_,
+                )
+
+                if i == n_batch - 1:
+                    # Compute the norm of the update of W in the last batch
+                    W_change = np.linalg.norm(self.W_ - W_last) / np.linalg.norm(W_last)
+
+            print(self.W_.shape, self.A_.shape, self.B_.shape)
+
+            self.benchmark_results_.append(
+                {
+                    "score": score,
+                    "W_change": W_change,
+                    "W_": self.W_.copy(),
+                    "A_": self.A_.copy(),
+                    "B_": self.B_.copy(),
+                }
+            )
+
+            if (W_change < self.tol) and (n_iter_ >= self.min_iter - 1):
+                break  # Stop if the change in W is smaller than the tolerance
+
+        # Update self.H_dict_ with the learned encoded vectors (activations)
+        self.H_dict_.update(zip(unq_X, unq_H))
+        return self
+
+
+class ModifiedGapEncoder(GapEncoder):
+    fitted_models_: List[ModifiedGapEncoderColumn]
+
+    @property
+    def benchmark_results_(self):
+        return self.fitted_models_[0].benchmark_results_
+
+    def _create_column_gap_encoder(self):
+        return ModifiedGapEncoderColumn(
+            ngram_range=self.ngram_range,
+            n_components=self.n_components,
+            analyzer=self.analyzer,
+            gamma_shape_prior=self.gamma_shape_prior,
+            gamma_scale_prior=self.gamma_scale_prior,
+            rho=self.rho,
+            rescale_rho=self.rescale_rho,
+            batch_size=self.batch_size,
+            tol=self.tol,
+            hashing=self.hashing,
+            hashing_n_features=self.hashing_n_features,
+            max_iter=self.max_iter,
+            init=self.init,
+            add_words=self.add_words,
+            random_state=self.random_state,
+            rescale_W=self.rescale_W,
+            max_iter_e_step=self.max_iter_e_step,
+        )
+
+
+benchmark_name = "gap_divergence"
+
+
+@monitor(
```

Comment:
I'd argue one of the reason I was able to pull out this benchmark so fast is thanks to this mini-framework, which takes care of the boilerplate (which is literally the same for all benchmarks), and is pretty well tested since we use it for all our benchmarks.
And it works well with your previous suggestion: saving progression on disk and hot-loading can be directly integrated in it, and other/future benchmarks will benefit it :smile:

Nonetheless, we agree it adds complexity, but I also think that's a necessary evil.